### PR TITLE
CLOUDP-167676: configure GOPROXY to use artifactory

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes

configure GOPROXY to use internal artifactory
_Jira ticket:_ CLOUDP-167676

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by tibulca** [comment]
wrong proxy url: https://spruce.mongodb.com/task/mongocli_master_code_health_compile_patch_3467e1a8688994cfd4873a6e3cfca2a554246f5d_641ae1ee32f417be4e11644b_23_03_22_11_09_35/logs?execution=0

only artifactory as GOPROXY:

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ variables:
       GOROOT: ${go_root}
       GOPATH: ${go_base_path}${workdir}
       ADD_PATH: "${go_bin}:${go_base_path}${workdir}/bin:${go_base_path}${workdir}/src/github.com/mongodb/mongodb-atlas-cli/bin"
+      GOPROXY: https://:@artifactory.corp.mongodb.com/artifactory/api/go/go,direct
```
Comment (tibulca) [inline build/ci/evergreen.yml:27]: @wtrocki can you please confirm this matches the recommendation you got from proxy.golang.org team, to use the internal artifactory and fallback to `direct`?

Comment (tibulca) [inline build/ci/evergreen.yml:27]: tried:
GOPROXY = wrong artifactory url: [evergreen task](https://spruce.mongodb.com/task/mongocli_master_code_health_compile_patch_3467e1a8688994cfd4873a6e3cfca2a554246f5d_641ae1ee32f417be4e11644b_23_03_22_11_09_35/logs?execution=0) (as expected, task failed)

GOPROXY = artifactory: [evergreen task](https://spruce.mongodb.com/task/mongocli_master_code_health_compile_patch_3467e1a8688994cfd4873a6e3cfca2a554246f5d_641ae2f7562343b64b0c3efd_23_03_22_11_14_00/logs?execution=0)

GOPROXY = artifactory + direct: [evergreen task](https://spruce.mongodb.com/task/mongocli_master_code_health_compile_patch_3467e1a8688994cfd4873a6e3cfca2a554246f5d_641ae429d1fe0798632f7421_23_03_22_11_19_07/logs?execution=0)

Comment (gssbzn) [reply build/ci/evergreen.yml:27]: can you please make this an expansion + project setting?

Comment (wtrocki) [reply build/ci/evergreen.yml:27]: Looks very good.

**Review by wtrocki — DISMISSED** [review]

**Review by gssbzn — APPROVED** [review]
🎉

File: build/ci/evergreen.yml (hunk 1 above)
Comment (tibulca) [reply build/ci/evergreen.yml]: thanks guys

**Pull request merged**